### PR TITLE
LNS: Handle transfer exceptions in RPC

### PR DESCRIPTION
**Pull request by Doy-lee**

### CODE REVIEW

File: src/wallet/wallet_rpc_server.cpp [hunk 1]
```diff
@@ -4353,21 +4363,31 @@ namespace tools
...
+                           res.tx_metadata,
+                           er);
+    }
+    catch (const std::exception &e)
+    {
+      handle_rpc_exception(std::current_exception(), er, WALLET_RPC_ERROR_CODE_GENERIC_TRANSFER_ERROR);
+      return false;
+    }
+
+    return true;
```
Comment (jagerman) [inline src/wallet/wallet_rpc_server.cpp]: This statement is unreachable: the code either returns from inside the try or inside the catch (or else, in theory, throws if given some non-std::exception-derived thrown value).

File: src/wallet/wallet_rpc_server.cpp [hunk 2]
```diff
@@ -4306,21 +4306,31 @@ namespace tools
...
-                         false /*blink*/,
-                         res.tx_hash,
-                         req.get_tx_hex,
-                         res.tx_blob,
-                         req.get_tx_metadata,
-                         res.tx_metadata,
-                         er);
+    try
+    {
+      fill_response(ptx_vector,
```
Comment (jagerman) [inline src/wallet/wallet_rpc_server.cpp]: Should this just be returning the `fill_response` value?  Is it possible for `fill_response` to return false?  (And if not, just make it a void function).

File: src/wallet/wallet_rpc_server.cpp (hunk 1 above)
Comment (Doy-lee) [reply src/wallet/wallet_rpc_server.cpp]: yeah removed.

File: src/wallet/wallet_rpc_server.cpp (hunk 2 above)
Comment (Doy-lee) [reply src/wallet/wallet_rpc_server.cpp]: Should be returning, matches the other code.

**Comment by Doy-lee** [comment]
Also added try/catch handling to some of the service node calls that weren't using it yet.

**Pull request merged**